src/WebJobs.Script.WebHost/WebJobsScriptHostService.cs

```diff
@@ -693,6 +694,17 @@ private void RegisterApplicationLifetimeEvents()
             _applicationLifetime.ApplicationStopping.Register(() =>
             {
                 Interlocked.Exchange(ref _applicationStopping, 1);
+                if (_environment.IsDrainOnApplicationStopping())
+                {
+                    var drainModeManager = _host?.Services.GetService<IDrainModeManager>();
+                    if (drainModeManager != null)
+                    {
+                        _logger.LogDebug("Application Stopping: enabling drain mode");
+                        drainModeManager.EnableDrainModeAsync(CancellationToken.None);
+                        Thread.Sleep(TimeSpan.FromMinutes(10));
```

Comment:
Do we need to sleep for 10 mins here regardless of the work being done in drain mode?

Reply:
This is supposed to be a stop gab solution until https://github.com/Azure/azure-functions-host/issues/7188 is done. it used to be 

```csharp
drainModeManager.EnableDrainModeAsync(CancellationToken.None).Wait();
```

as I assumed that task will finish when all the listeners have stopped and all functions finished, but @soninaren explained that this won't work (works for storage queues, but not other triggers and I happen to just test with queues) 

The sleep here is intended as a timeout from the time we receive SIGTERM, giving any function that started at least 10 minutes to finish. It's not a great solution, but should match consumption limit.

Reply:
Can you mark this code TEMP then, with a pointer to the work item https://github.com/Azure/azure-functions-host/issues/7188?